samples/matter/lock/src/app_event.h

```diff
@@ -6,44 +6,42 @@
 
 #pragma once
 
-#include <cstdint>
+#include "event_manager.h"
+
+enum class AppEventType : uint8_t { None = 0, NUSCommand, LockEvent, ThreadWiFiSwitch };
+
+struct AppEvent : public Event {
+	AppEvent()
+		: Event(EventSource::Application, static_cast<uint8_t>(AppEventType::None), nullptr, this,
+			sizeof(*this))
+	{
+	}
+	AppEvent(AppEventType type, EventHandler handler = nullptr)
+		: Event(EventSource::Application, static_cast<uint8_t>(type), handler, this, sizeof(*this))
+	{
+	}
+
+	/**
+	 * @brief Construct a new App Event object from context
+	 *
+	 * User must ensure that context exists, is defined and has the same size as the AppEvent class,
```

Comment:
Isn't it that in the current solution, the context must be exactly of AppEvent type?